ssh/controlmaster.go

```diff
@@ -0,0 +1,233 @@
+// Copyright 2013 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ssh
+
+import (
+	"encoding/binary"
+	"errors"
+	"io"
+	"net"
+)
+
+// controlMasterMessage covers multiple CM packets that have no data
+type controlMasterMessage struct {
+	PacketType uint32 // MUX_C_ALIVE_CHECK|MUX_C_PROXY|MUX_S_PROXY|MUX_S_OK
+	RequestID  uint32
+}
+
+// controlMasterAliveMessage is a response to a MUX_C_ALIVE_CHECK request
+type controlMasterAliveMessage struct {
+	PacketType uint32 // MUX_S_ALIVE
+	RequestID  uint32
+	ServerPID  uint32
+}
+
+// controlMasterHelloMessage is used to ask/send the protocol version
+type controlMasterHelloMessage struct {
+	PacketType      uint32 // MUX_MSG_HELLO
+	ProtocolVersion uint32
+}
+
+// DialControlMaster starts a client connection to the given SSH server.
```

Comment:
You probably want to document here an below that this is an OpenSSH extension and may not be implemented by all servers.  Even better would be explaining (e.g. by giving an example command) how to set up an openssh server in control mode, particularly if you don't implement this in the Go server.